pkgs/applications/misc/blender/default.nix

```diff
@@ -225,6 +225,16 @@ stdenv.mkDerivation (finalAttrs: rec {
             --render-frame 1
         done
       '';
+
+      cudaAvailable = runCommand
+      "blender-cuda-available"
+      {
+        nativeBuildInputs = [ finalAttrs.finalPackage ];
+        requiredSystemFeatures = [ "cuda" ];
```

Comment:
Putting derivations in ``passthru`` can lead to splicing issues.
What if we compile on say ``aarch64`` but need to execute this on ``x86_64``?
Would ``preferLocalBuild`` alleviate this issue?

Reply:
Oh, that's a very interesting question, I'll have to think about this. But I'm inclined to think this isn't a blocker, because do we even run tests non-native builds at this point? We disable checkPhases for python packages, and I'm not sure there's any automation in place to pick up the passthru tests from cross instantiations.

Pondering at the question though:

1. The semantics of the feature is that it asks for cuda devices to be available during the build, and driver usable on the build platform
2.If we instantiate `import nixpkgs { system = "aarch64-linux"; crossSystem = "x86_64-linux"; }` (I find the opposite direction more likely?) and try to access `blender.cudaAvailable`, we want
2.1 `blender`'s `buildPlatform` to be `aarch64-linux` and `x86_64-linux` the `hostPlatform`
2.2 `cudaAvailable`'s `buildPlatform` to be `x86_64-linux`
3. Then Nix will happily build `blender` on the `aarch64-linux`, and ask for a `x86_64-linux` remote builder for `cudaAvailable`. Or refuse to build if one's not available

Reply:
Splicing aside, `pkgsHostTarget.callPackage ./test-gpu.nix { }` should do the trick

Reply:
But how does splicing treat passthru? I've no idea

Reply:
> But how does splicing treat passthru? I've no idea

This is the specific issue. https://github.com/NixOS/nixpkgs/issues/211340
``pkgsHostTarget.callPackage`` should work.
Are we able to define the ``buildPlatform`` for ``cudaAvailable``, or for that matter, for ``allowedPatterns``?

Reply:
`allowedPatterns` are just data, there's nothing to build in there.

As for "defining the platform for `cudaAvailable`", isn't it exactly what we do if we `pkgs<X><Y>.callPackage`? Because that cross package set should contain its own `stdenv` with shifted build- and hostPlatforms. By the way, I was probably wrong about `pkgsHostTarget`, it should have been `pkgsTargetTarget` instead and I'm not sure if that actually works anyway:

```
❯ nix eval -f '<nixpkgs>' --arg crossSystem '{ config = "aarch64-unknown-linux-gnu"; }' pkgsHostTarget.buildPlatform.system
"x86_64-linux"
❯ nix eval -f '<nixpkgs>' --arg crossSystem '{ config = "aarch64-unknown-linux-gnu"; }' pkgsTargetTarget.buildPlatform.system
error: attribute 'buildPlatform' in selection path 'pkgsTargetTarget.buildPlatform.system' not found
```

Regardless, figuring out how to make passthru tests work in cross scenarios is way out of scope for this PR, there's the entire `python3Packages` full of passthroughs which still  doesn't care about non-native builds